File: ProblemSolve/404.cpp
/**********************
 * Sum of Left Leaves *
 **********************/

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    int sumOfLeftLeaves(TreeNode* root, bool isLeft = false) {
        if(root == NULL){
            return 0;
        }
        int sum = 0;
        
        //leaf이면서 left 노드인지 확인
        if(root -> left == NULL && root -> right == NULL && isLeft){
            sum = root -> val;
        }
        
        //모든 tree를 순회하면서 left leaf 값을 더한다
        sum = sum + sumOfLeftLeaves(root -> left, true) + sumOfLeftLeaves(root -> right, false);
        
        return sum;
    }
};

File: ProblemSolve/404.cs
/**********************
 * Sum of Left Leaves *
 **********************/

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    public int SumOfLeftLeaves(TreeNode root, bool isLeft = false) {
        if(root == null){
            return 0;
        }
        
        int ans = 0;
        
        //left leaf면 정답에 더해준다
        do{
            if(root.left != null){
                break;
            }
            
            if(root.right != null){
                break;
            }
            
            if(!isLeft){
                break;
            }
            
            ans += root.val;
            return ans;
        }while(false);
        
        ans = SumOfLeftLeaves(root.left, true) + SumOfLeftLeaves(root.right, false);
        
        return ans;
    }
}